user-operator/src/main/java/io/strimzi/operator/user/operator/batching/AbstractBatchReconciler.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator.batching;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ *
+ */
+public abstract class AbstractBatchReconciler<T> implements Runnable {
```

Comment:
By implementing `Runnable` you expose a `run()` method to clients which they should never call (you expect them to instead follow the pattern of instantiate, `start()`, `enqueue()` ... `enqueue()`, `stop()`). So it would be better from an API design PoV to use a private inner class that implements Runnable to avoid ending up with a public `run()` that you never want anyone to call.